firebase-vscode/src/core/emulators.ts

```diff
@@ -156,115 +151,98 @@ export class EmulatorsController implements Disposable {
-          };
-          await emulatorsStart(uiSelections);
-          this.emulators.value = {
-            status: "running",
-            infos: {
-              uiUrl: getEmulatorUiUrl(),
-              displayInfo: listRunningEmulators(),
-            },
-          };
-
-          vscode.window.showInformationMessage(
-            "Firebase Extension: Emulators started successfully"
-          );
-
-          // data connect specifics; including temp logging implementation
-          if (
-            listRunningEmulators().filter((emulatorInfos) => {
-              emulatorInfos.name === Emulators.DATACONNECT;
-            })
-          ) {
-            const dataConnectEmulatorDetails = getEmulatorDetails(
-              Emulators.DATACONNECT
-            );
-
-            dataConnectEmulatorDetails.instance.stdout?.on("data", (data) => {
-              emulatorOutputChannel.appendLine("DEBUG: " + data.toString());
-            });
-            dataConnectEmulatorDetails.instance.stderr?.on("data", (data) => {
-              if (data.toString().includes("Finished reloading")) {
-                vscode.commands.executeCommand("fdc-graphql.restart");
-                vscode.commands.executeCommand(
-                  "firebase.dataConnect.executeIntrospection"
-                );
-              } else {
-                emulatorOutputChannel.appendLine("ERROR: " + data.toString());
-              }
-            });
-          }
-        } catch (e) {
-          this.emulators.value = {
-            status: "stopped",
-            infos: undefined,
-          };
-
-          vscode.window.showErrorMessage(
-            "Firebase Extension: Emulators start failed - " + e
-          );
-        }
-        progress.report({ increment: 100 });
-      }
-    );
+  async waitEmulators() {
+    await this.pendingEmulatorStart;
   }
 
-  async stopEmulators() {
-    return vscode.window.withProgress(
-      {
-        location: vscode.ProgressLocation.Window,
-        cancellable: false,
-        title: "Stopping emulators",
-      },
-      async (progress) => {
-        progress.report({ increment: 0 });
-
+  async startEmulators() {
+    this.emulatorStatusItem.text = "$(loading~spin) Emulators: Starting...";
+    this.emulatorStatusItem.show();
+    this.emulatorStatusItem.backgroundColor = undefined;
+    this.emulators.value = {
+      status: "starting",
+      infos: this.emulators.value.infos,
+    };
+
+    const currentOp = (this.pendingEmulatorStart = new Promise(async () => {
+      try {
+        await emulatorsStart(this.uiSelections.value);
         this.emulators.value = {
-          status: "stopping",
-          infos: this.emulators.value.infos,
+          status: "running",
+          infos: {
+            uiUrl: getEmulatorUiUrl(),
+            displayInfo: listRunningEmulators(),
+          },
         };
-        await stopEmulators();
+        this.emulatorStatusItem.text = "Emulators: running";
```

Comment:
(Dataconnect Logo) Toolkit: running

Reply:
Ignore the above.

We only want to show "(FDC Logo) Emulators: running" when the user has explicitly connected to psql